src/openslide-vendor-dicom.c

```diff
@@ -0,0 +1,943 @@
+  label_types,
+  overview_types
+};
+static const struct allowed_types associated_types = {
+  associated_type_strings, 
+  sizeof(associated_type_strings) / sizeof(associated_type_strings[0])
+};
+
+G_DEFINE_AUTOPTR_CLEANUP_FUNC(DcmFilehandle, dcm_filehandle_destroy)
+G_DEFINE_AUTOPTR_CLEANUP_FUNC(DcmDataSet, dcm_dataset_destroy)
+
+static void set_gerror_from_dcm_error(GError **err, DcmError **dcm_error)
+{
+  char *msg = g_strdup_printf("libdicom %s: %s - %s",
+    dcm_error_code_str(dcm_error_code(*dcm_error)),
+    dcm_error_summary(*dcm_error),
+    dcm_error_message(*dcm_error));
+  g_set_error(err, OPENSLIDE_ERROR, OPENSLIDE_ERROR_FAILED, "%s", msg);
+  g_free(msg);
+  dcm_error_clear(dcm_error);
+}
+
+#ifdef DEBUG
+static void print_file(struct dicom_file *f) {
+  printf("file:\n" );
+  printf("  filename = %s\n", f->filename);
+  printf("  filehandle = %p\n", f->filehandle);
+  printf("  metadata = %p\n", f->metadata);
+  printf("  meta = %p\n", f->meta);
+  printf("  bot = %p\n", f->bot);
+}
+
+static void print_level(struct level *l) {
+  printf("level:\n" );
+  print_file(l->file);
+  printf("  base.downsample = %g\n", l->base.downsample);
+  printf("  grid = %p\n", l->grid);
+  printf("  format = %d\n", l->image_format);
+  printf("  image_width = %d\n", l->image_width);
+  printf("  image_height = %d\n", l->image_height);
+  printf("  tile_w = %d\n", l->tile_w);
+  printf("  tile_h = %d\n", l->tile_h);
+  printf("  num_frames = %d\n", l->num_frames);
+  printf("  tiles_across = %d\n", l->tiles_across);
+  printf("  tiles_down = %d\n", l->tiles_down);
+}
+
+static void print_frame(DcmFrame *frame) {
+  printf("value = %p\n", dcm_frame_get_value(frame));
+  printf("length = %u bytes\n", dcm_frame_get_length(frame));
+  printf("rows = %u\n", dcm_frame_get_rows(frame));
+  printf("columns = %u\n", dcm_frame_get_columns(frame));
+  printf("samples per pixel = %u\n",
+         dcm_frame_get_samples_per_pixel(frame));
+  printf("bits allocated = %u\n", dcm_frame_get_bits_allocated(frame));
+  printf("bits stored = %u\n", dcm_frame_get_bits_stored(frame));
+  printf("high bit = %u\n", dcm_frame_get_high_bit(frame));
+  printf("pixel representation = %u\n",
+         dcm_frame_get_pixel_representation(frame));
+  printf("planar configuration = %u\n",
+         dcm_frame_get_planar_configuration(frame));
+  printf("photometric interpretation = %s\n",
+         dcm_frame_get_photometric_interpretation(frame));
+  printf("transfer syntax uid = %s\n",
+         dcm_frame_get_transfer_syntax_uid(frame));
+}
+#endif /*DEBUG*/
+
+static bool dicom_detect(const char *filename,
+                         struct _openslide_tifflike *tl, 
+                         GError **err) {
+  // reject TIFFs
+  if (tl) {
+    g_set_error(err, OPENSLIDE_ERROR, OPENSLIDE_ERROR_FAILED,
+                "Is a TIFF file");
+    return false;
+  }
+
+  // verify existence
+  GError *tmp_err = NULL;
+  if (!_openslide_fexists(filename, &tmp_err)) {
```

Comment:
Is this necessary, or can we just let the open call fail if the file is missing?

Reply:
OK, gone!